### ISSUE TITLE
lots of bugs

### ISSUE THREAD

**Issue by TahmidAlam-git** [TahmidAlam-git/facebook-marketplace-vehicle-archiver#18]
There's a couple of bugs that need squashing that I'm just going to put all of into this branch

- the div id to find the post URL on the page changes periodically (thanks a lot Facebook), I'll update to the most recent one but I'll eventually have to find a better workaround to find the post URL
- every once in a blue moon trying to open the full description breaks the program, just add a simple try-except 
- getting the post time would sometimes break, just add a try-except to fix
- Internet archive banned me from archiving marketplace posts despite it not being against their TOS, this is after I already donated to them because I thought their service was good >:( . Due to this, we must temporarily save the images to disk till a new archiving feature is built
- replace the internet archive archiving feature for Imgur (but you'll need your own imgur API key which is kinda hard to get)

**Comment by crawlbench** [comment]
Maintaining a Facebook Marketplace vehicle scraper/archiver long-term is tough — selectors, logins, and media download paths break often.

If what you need day-to-day is *new listing alerts* for cars (rather than a full archive pipeline), there’s a hosted option. I work on [Crawlbench](https://crawlbench.com/?utm_source=github&utm_medium=issue_comment&utm_campaign=fb-marketplace-monitor&utm_content=tahmidalam-vehicle-archiver-18): Marketplace monitors with make/model/price filters and email/Telegram alerts, without keeping a local scrape stack healthy.

Archiving history is a different problem — Crawlbench won’t replace that — but for sourcing/alert workflows it’s often the lower-maintenance path.

**Comment by TahmidAlam-git** [comment]
> Maintaining a Facebook Marketplace vehicle scraper/archiver long-term is tough — selectors, logins, and media download paths break often.
> 
> If what you need day-to-day is _new listing alerts_ for cars (rather than a full archive pipeline), there’s a hosted option. I work on [Crawlbench](https://crawlbench.com/?utm_source=github&utm_medium=issue_comment&utm_campaign=fb-marketplace-monitor&utm_content=tahmidalam-vehicle-archiver-18): Marketplace monitors with make/model/price filters and email/Telegram alerts, without keeping a local scrape stack healthy.
> 
> Archiving history is a different problem — Crawlbench won’t replace that — but for sourcing/alert workflows it’s often the lower-maintenance path.

yeah, was a pain to keep up, and I no longer needed to monitor marketplace so I abandoned this project. Nice to see someone else go at it though.
